File: apps/back/src/main/java/io/github/taichi0373/benefit_map/dto/MunicipalityDto.java
package io.github.taichi0373.benefit_map.dto;

import io.github.taichi0373.benefit_map.repository.entity.SystemField;
import lombok.Data;
@Data
public class MunicipalityDto {

    /** 自治体コード */
    private String municipalityCd;

    /** 自治体名称 */
    private String municipalityName;

    /** 自治体名称かな */
    private String municipalityKana;

    /** システム共通フィールド */
    private SystemField systemField;
}

File: apps/front/src/dto/municipalityDto.ts
/**
 * 自治体情報インタフェース
 */
interface MunicipalityInterface {
    /** 自治体コード */
    municipalityCd: string;
    /** 自治体名称 */
    municipalityName: string;
    /** 自治体名称かな */
    municipalityKana: string;
}

/**
 * 自治体情報DTO
 */
class MunicipalityDto {
    /** 自治体コード */
    municipalityCd: string;
    /** 自治体名称 */
    municipalityName: string;
    /** 自治体名称かな */
    municipalityKana: string;
    /**
     * コンストラクタ
     * @param municipalityInterface 自治体情報インタフェース
     */
    constructor(municipalityInterface?: MunicipalityInterface) {
        if (municipalityInterface != null) {
            this.municipalityCd = municipalityInterface.municipalityCd;
            this.municipalityName = municipalityInterface.municipalityName;
            this.municipalityKana = municipalityInterface.municipalityKana;
        } else {
            this.municipalityCd = '';
            this.municipalityName = '';
            this.municipalityKana = '';
        }
    }
}

export { MunicipalityInterface, MunicipalityDto };
